### PR TITLE
moved development docu to sub-folder

### PR DESCRIPTION
**Pull request by cheld**
I moved the development documentation to the doc subfolder.

More and more end-users come across the project and try build the project in order to try it out.

We get a lot of issues with users failing on this attempt. However, the documentation is not intended for end-users so I moved it to a sub folder.

### CODE REVIEW

**Comment by bryk** [comment]
Yes, this is very good change.

**Comment by bryk** [comment]
How about adding a link from the front page to the development docs? It is hard to find otherwise.

File: README.md [hunk 1]
```diff
@@ -9,114 +9,6 @@ itself.
 
 The Dashboard is currently under active development and is not ready for production use (yet!).
 
-## Architecture Overview
```
Comment (luxas) [inline README.md]: Maybe there should be a link to getting started?

**Comment by bryk** [comment]
Now looks really nice. I'm merging. LGTM

**Pull request merged**